### ISSUE TITLE
[Fallout 76] Strange Colors

### ISSUE THREAD

**Issue by atm0915** [doitsujin/dxvk#779]
Colors in Fallout 76 seem to render very strangely.
### Software information
- Fallout 76.exe launched standalone
- It was downloaded from the Bethesda Launcher on a Windows installation.

### Screenshots
 ![Example 1](https://i.imgur.com/r5u8Txe.jpg)
 ![Example 2](https://i.imgur.com/oEivzIH.jpg)

### System information
- GPU: GTX 1080
- Driver: Nvidia 396.54.09
- Wine version: 3.20
- DXVK version: 0.92

### Apitrace file(s)
- https://drive.google.com/open?id=1bbXqQBrWNJ6nDBsW0ni6l_plIAPmnWgq

### Log files
- d3d11.log: https://pastebin.com/hN20d7Bj
- dxgi.log: https://pastebin.com/9gnQuq7J

**Comment by beniwtv** [comment]
I have not noticed this on an AMD RX480 using Mesa. If it helps narrow down the issue, I could make an API trace too. 

The only issue I see is that some grass appears to have a "film-grain" effect over it - but I am not sure it's DXVK related at all as some Windows friends also report weirdness with grass.

btw @atm0915 Bethesda Launcher works in Wine (I only tried -staging), just needs corefonts to boot, and every time you want to download/update a game you have to rename a file manually (see WineHQ's entry for the launcher, it's explained there).

**Comment by doitsujin** [comment]
Thanks for providing the apitrace. Same thing happens when replaying the trace on my RX 480 - I hope it's not baked into the trace, because that would be insanely hard to debug. Will investigate next week.

**Comment by atm0915** [comment]
I am open to do anything to help you debug if you need me to.

**Comment by K0bin** [comment]
https://i.imgur.com/h5R38S2.jpg

Can reproduce with D3D11 :(

**Comment by atm0915** [comment]
/u/Nugaryal reported the same issue in this [reddit post](https://www.reddit.com/r/wine_gaming/comments/9yt95u/fallout_76/ea4f0qa/), he gave me his specs to add here if they are useful. 
RTX 2080 TI 
Nvidia driver: 415.18 
Wine: 3.20 
DXVK: 0.92

**Comment by jcrollard84** [comment]
Same issue. Using the same wine / Nvidia driver as atm. My card is NVIDIA 1060. The game works fine but everything is very red.

**Comment by jcrollard84** [comment]
> Thanks for providing the apitrace. Same thing happens when replaying the trace on my RX 480 - I hope it's not baked into the trace, because that would be next to impossible to debug. Will investigate next week.

Any luck? I know it's probably too soon but let me know if you need someone to test / debug anything and I can help.

**Comment by enthalpie** [comment]
Hi there, I like to help too but I dont get past the login screen. There is no sound and I can log in but it says: servers are down for maintenance; what was not true and I double checked in Windows. I have the same setup like you, except for the graphics card:

    GPU: GTX 1060
    Driver: Nvidia 396.54.09
    Wine version: 3.20
    DXVK version: 0.92

I have a clean wine-staging install, what else did you add? Corefonts and xact maybe?
Any tipps appreciated, thank you in advance :)

**Comment by Riesi** [comment]
Alright I got 2 apitraces from Fallout 76 on windows here is the first one, because the 2nd stopped uploading and I am still reuploading. The game crashed while recording this one, but the trace replayed fine on the windows machine. The trace only crashed apitrace at the same point the game crashed. I will post the 2nd trace when its done uploading in approximately 1h.

Windows trace:
https://drive.google.com/open?id=12152NFduUTqVmBajBPi1KNbP6yU1AU0T

**Comment by Riesi** [comment]
Here is the 2nd trace:
https://drive.google.com/open?id=1szYvyJc7GrZrHgdXtvdCeps_MDWtTnyW

**Comment by enthalpie** [comment]
I have solved my issue, I always got the error 'NoRegionPing' upon connecting in Fallout. I installed Visual C++ Redistributable for Visual Studio 2015 as an exe in the wineprefix and then it worked. 

So now I have the same color glitches, I tried to make apitraces dont know if it worked tho. 

![bildschirmfoto vom 2018-12-07 00-49-14](https://user-images.githubusercontent.com/3176329/49619334-76f90800-f9bc-11e8-8d62-eebfd04b8a2e.png)

dxgi.log - https://pastebin.com/VTirFTcs

Traces:
https://drive.google.com/file/d/1szYvyJc7GrZrHgdXtvdCeps_MDWtTnyW/view?usp=sharing

**Comment by doitsujin** [comment]
So, two things I discovered:
- The apitraces that @Riesi posted render correctly
- The apitraces recorded with DXVK don't, but apitrace actually doesn't update some constant buffers correctly. Not sure if that is in any way related to the rendering issues we're seeing.

**Comment by doitsujin** [comment]
@enthalpie @atm0915 can you test 9572425912c2ee07ee112d8fe75a817c0f82b85c please? Here's a build in case you need it:

[dxvk-master.tar.gz](https://github.com/doitsujin/dxvk/files/2657282/dxvk-master.tar.gz)

**Comment by enthalpie** [comment]
Thanks for your effort, I have tried your prebuilt fix but it is still all very red. The GPU tho gets taxed lesser, before the game maxed out my VRAM of 6 GB, now it stays within reasonable ranges.

**Comment by enthalpie** [comment]
I may be on to something: in the wineprefix under /users/Username/My Documents/My Games/Fallout76 I deleted both ini files and the saves flder. Then started he game and now the raphics ar perfect. I changed the Fallout.ini as well, but I think deleting those files already does the trick. The graphics are set to ultra and run like in Windows and with 60 fps, just perfect. :)

Only downside is, the game and music hangs every minute or so for 5 seconds and then goes on. That stuttering is really hard. Will try a few things like switching of esync

**Comment by jcrollard84** [comment]
> I may be on to something: in the wineprefix under /users/Username/My Documents/My Games/Fallout76 I deleted both ini files and the saves flder. Then started he game and now the raphics ar perfect. I changed the Fallout.ini as well, but I think deleting those files already does the trick. The graphics are set to ultra and run like in Windows and with 60 fps, just perfect. :)
> 
> Only downside is, the game and music hangs every minute or so for 5 seconds and then goes on. That stuttering is really hard. Will try a few things like switching of esync.
> 
> Edit: that's it, no esync did the trick and it runs perfect now on ultra settings with 60 fps. Thank you very much doitsujin, great work the Dxvk layer :). Danke Alter!
> 
> My setup now:
> Lutris
> Wine: tkg-protonified-3.21-x86_64
> DXVK: 0.93
> Esync disabled
> Driver Nvidia 396.54. 09

Interesting. Tried deleting those files but no luck. Same issue still. I wonder what's different?
I'm using the same wine and dxvk version.

**Comment by jcrollard84** [comment]
> I may be on to something: in the wineprefix under /users/Username/My Documents/My Games/Fallout76 I deleted both ini files and the saves flder. Then started he game and now the raphics ar perfect. I changed the Fallout.ini as well, but I think deleting those files already does the trick. The graphics are set to ultra and run like in Windows and with 60 fps, just perfect. :)
> 
> Only downside is, the game and music hangs every minute or so for 5 seconds and then goes on. That stuttering is really hard. Will try a few things like switching of esync.
> 
> Edit: that's it, no esync did the trick and it runs perfect now on ultra settings with 60 fps. Thank you very much doitsujin, great work the Dxvk layer :). Danke Alter!
> 
> My setup now:
> Lutris
> Wine: tkg-protonified-3.21-x86_64
> DXVK: 0.93
> Esync disabled
> Driver Nvidia 396.54. 09

Enthal, I'm pretty sure this has to do with Fallout76Prefs.ini

I've been messing with it and definitely getting changes. A few edits and everything was green instead of red, for me. Can you post your prefs file here if it's working well for you?

**Comment by jcrollard84** [comment]
Sorry for another post, but I do believe I got it working on my end!

First create a file in your My Games/Fallout76 directory called Fallout76Custom.ini

Then put this in it:

[ScreenSplatter]

bBloodSplatterEnabled = 0

[Grass]
iMinGrassSize = 80
bAllowCreateGrass = 0

[ImageSpace]
bDoRadialBlur = 0

[Weather]
bPrecipitation = 0
bFogEnabled = 0
bRainOcclusion = 0
bWetnessOcclusion = 0

[Enlighten]
bEnableEnlighten = 0

Graphics are perfect after that, and works with Esync (for me).

Found these tips on another site for someone who was having graphic issues on windows..

**Comment by enthalpie** [comment]
I have changed 3 or 4 values myself in the ini's that could be it. If I compare mine to your values, I have FogDisabled and enableEnlighten set to zero, the rest I have set to one. So maybe it is one of those two causing the problem.
After deleting the Fallout76prefs.ini I figured, the one newly created was much shorter. Here are my ini's. The Fallout76.ini script is created from the main game folder, so it should be changed there and after the changes be deleted in the users/username/My Documents/My Games/Fallout76 folder

[Fallout76.zip](https://github.com/doitsujin/dxvk/files/2659576/Fallout76.zip)

**Comment by enthalpie** [comment]
The solution is:
bFogEnabled=0
bEnableEnlighten=0

In Fallout76.ini in the main game folder. One of the two or both. 

Problem solved :), just the framedrops still make it unplayable for me

**Comment by Makaronelle** [comment]
Have the same issue here with a GTX 1060 on 396.54.09, here's my logs:
[https://gist.githubusercontent.com/Makaronelle/ea783c4e08d4cc7369ad08ee124ff84f/raw/b9b9077437f9c0bf2ace7af78689241a75e86da2/terminal.log](url)
[https://gist.githubusercontent.com/Makaronelle/0828834ad0e9b0f9eb08b1fd0cd0d889/raw/a75943a09cafd28525652de504181d6a2b458796/Fallout76_d3d11.log](url)

Hope it may help !

**Comment by enthalpie** [comment]
> Have the same issue here with a GTX 1060 on 396.54.09, here's my logs:
> https://gist.githubusercontent.com/Makaronelle/ea783c4e08d4cc7369ad08ee124ff84f/raw/b9b9077437f9c0bf2ace7af78689241a75e86da2/terminal.log
> https://gist.githubusercontent.com/Makaronelle/0828834ad0e9b0f9eb08b1fd0cd0d889/raw/a75943a09cafd28525652de504181d6a2b458796/Fallout76_d3d11.log
> 
> Hope it may help !

Navigate to the directory, where your Fallout76.exe is located and log for the Fallout76.ini. Open it with a Texteditor and look for " bEnableEnlighten = 1". change the 1 to a zero and save the file. After a Fallout Update, you may have to do it again. That should fix the red tint for the time being.

I have an error now that is really bugging me, but it seems to be windows-related. DXVK is working so good that I can already suffer Fallout errors, normally only Windows users have. :D Every now and then, I get framedrops to zero and the game hangs for five seconds. That is pretty bad in fights and almost always results in death. 
The windows people didnt seem to have fixed it so far from what I have read and its persisting since Fallout 4. Disabling VSync which locks the framerate in Fallout76 is an idea and another one is a memory leak regarding a two small pagefile. I dont know tho if thats with me and 16 GB RAM a real issue. I have reverted the NvAPI workaround and disabled napi manually but that only had an effect on micro freezes within Battlefield 5 (that runs Platinum now). In Fallout the error persits.

I will play a bit with settings on the weekend.

**Comment by Makaronelle** [comment]
> > Have the same issue here with a GTX 1060 on 396.54.09, here's my logs:
> > https://gist.githubusercontent.com/Makaronelle/ea783c4e08d4cc7369ad08ee124ff84f/raw/b9b9077437f9c0bf2ace7af78689241a75e86da2/terminal.log
> > https://gist.githubusercontent.com/Makaronelle/0828834ad0e9b0f9eb08b1fd0cd0d889/raw/a75943a09cafd28525652de504181d6a2b458796/Fallout76_d3d11.log
> > Hope it may help !
> 
> Navigate to the directory, where your Fallout76.exe is located and look for the Fallout76.ini. Open it with a Texteditor and look for " bEnableEnlighten = 1". change the 1 to a zero and save the file. After a Fallout Update, you may have to do it again. That should fix the red tint for the time being.
> 
> I have an error now that is really bugging me, but it seems to be windows-related. DXVK is working so good that I can already suffer Fallout errors, normally only Windows users have. :D Every now and then, I get framedrops to zero and the game hangs for five seconds. That is pretty bad in fights and almost always results in death.
> The windows people didnt seem to have fixed it so far from what I have read and its persisting since Fallout 4. Disabling VSync which locks the framerate in Fallout76 is an idea and another one is a memory leak regarding a two small pagefile. I dont know tho if thats with me and 16 GB RAM a real issue. I have reverted the NvAPI workaround and disabled nvapi manually but that only had an effect on micro freezes within Battlefield 5 (that runs Platinum now). In Fallout the error persits.
> 
> I will play a bit with settings on the weekend.

Oh, don't worry, I already made my little investigation on that issue a couple of days ago so I could make a workable Lutris script for both AMD and Nvidia. Just posting logs to help solving this.

It may be easier if you simply put your custom settings in a Fallout76Custom.ini in your game "save" folder to prevent it from behind reseted every now and then with an update, directly editing the root folder's inis may give mixed result, plus it's cleaner that way !

**Comment by enthalpie** [comment]
I was able to get rid of the frame drops or better micro freezes. The value "iPresentInterval" in users/username/my Documents/My Games/Fallout76Prefs.ini needs to be set to zero so VSync is disabled. The game runs perfect now for me at 80 fps, 1080p with maximum settings.

Juhu :)

This is prolly game engined related tho, cause the engine is crap.

**Comment by Taraluktus** [comment]
Is the red tinting still a thing?
Just bought the game yesterday, installed it in Windows, and started it with a fresh prefix (corefonts, xact, vcrun2015). Didn't change the .INI files, but the game runs nevertheless without red tinting.

Maybe someone who previously had red tinting with an older game version could test this, with bEnableEnlighten = 1 ?

**Comment by beniwtv** [comment]
@Taraluktus What GPU do you have? For me, on AMD the red tinting also doesn't happen.

**Comment by Taraluktus** [comment]
LLVM 8, Mesa 19-dev, with RADV (RX 480)
Ah, I've misread your comment from Nov 23. Really seems to be NVIDIA only.

**Comment by ZachBacon** [comment]
Perhaps in a dxvk conf for the game, spoof an amd card and see about it. Just out of curiosity.

**Comment by Riesi** [comment]
AMD is spoofed by default since commit 5d49644cdc071866c5556997a35bd946d7724457

**Comment by ZachBacon** [comment]
Oof. There goes that idea then.

**Comment by jarrard** [comment]
Ran faster using proton beta with esync, however it stalled just at the point of rendering the world, and some pixels appear in the black darkness but CPU tanks to 0%, so I must be missing something critical, shrug. Maybe having no xact or faudio is causing this,

**Comment by Taraluktus** [comment]
@jarrard Have a look at https://appdb.winehq.org/objectManager.php?sClass=version&iId=37459
Issues concerning WINE should probably go there, since game (not launcher!) runs pretty well&stable with DXVK.

**Comment by jarrard** [comment]
SIGH,

I had a custom dxgi.dll file in my fo76 folder this whole time, facepalm...... 

Still yet to get sound working but its a start.

**Comment by jarrard** [comment]
Water does show as white for me so far. Seems odd.

**Comment by jarrard** [comment]
Just thought I mention this here and see if anyone else has this issue.

There is instances where light sources flicker in this game, I don't remember those happening under windows, basically cabin lights etc will flicker depending on angle you look at them. Might be a issue?

**Comment by ZachBacon** [comment]
> Just thought I mention this here and see if anyone else has this issue.
> 
> There is instances where light sources flicker in this game, I don't remember those happening under windows, basically cabin lights etc will flicker depending on angle you look at them. Might be a issue?

Nah, that actually happens outside of dxvk using the native renderer even on windows

**Comment by enthalpie** [comment]
On Nvidia I had to set bEnableEnlighten=0 to get rid of the red tint and iPresentInterval=0 to get rid of vsync, cause that caused micro freezes every 10 seconds. Then I had to use libstrangle from gitlab to limit the fps otherwise the game would crash.

Tried on AMD now with a Vega 64 and it runs out of the box. Only have to set bBackgroundMouse=0 to capture the mouse but Nvidia had that wine issue too. 

The lutris installer btw works flawless, game runs Platinum for me.
